src/Umbraco.Core/Cache/AppCaches.cs

```diff
@@ -0,0 +1,99 @@
+        /// </summary>
+        public AppCaches(System.Web.Caching.Cache cache)
+            : this(
+                new WebCachingAppCache(cache),
+                new DictionaryCacheProvider(),
+                new HttpRequestAppCache(),
+                new IsolatedCaches(t => new ObjectCacheAppCache()))
+        { }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AppCaches"/> with cache providers.
+        /// </summary>
+        public AppCaches(
+            IAppPolicedCache runtimeCache,
+            IAppCache staticCacheProvider,
+            IAppCache requestCache,
+            IsolatedCaches isolatedCaches)
+        {
+            RuntimeCache = runtimeCache ?? throw new ArgumentNullException(nameof(runtimeCache));
+            StaticCache = staticCacheProvider ?? throw new ArgumentNullException(nameof(staticCacheProvider));
+            RequestCache = requestCache ?? throw new ArgumentNullException(nameof(requestCache));
+            IsolatedCaches = isolatedCaches ?? throw new ArgumentNullException(nameof(isolatedCaches));
+        }
+
+        /// <summary>
+        /// Gets the special disabled instance.
+        /// </summary>
+        /// <remarks>
+        /// <para>When used by repositories, all cache policies apply, but the underlying caches do not cache anything.</para>
+        /// <para>Used by tests.</para>
+        /// </remarks>
+        public static AppCaches Disabled { get; } = new AppCaches(NoAppCache.Instance, NoAppCache.Instance, NoAppCache.Instance, new IsolatedCaches(_ => NoAppCache.Instance));
+
+        /// <summary>
+        /// Gets the special no-cache instance.
+        /// </summary>
+        /// <remarks>
+        /// <para>When used by repositories, all cache policies are bypassed.</para>
+        /// <para>Used by repositories that do no cache.</para>
+        /// </remarks>
+        public static AppCaches NoCache { get; } = new AppCaches(NoAppCache.Instance, NoAppCache.Instance, NoAppCache.Instance, new IsolatedCaches(_ => NoAppCache.Instance));
+
+        /// <summary>
+        /// Gets the per-request cache.
+        /// </summary>
+        /// <remarks>
+        /// <para>The per-request caches works on top of the current HttpContext items.</para>
+        /// fixme - what about non-web applications?
+        /// </remarks>
+        public IAppCache RequestCache { get; }
+
+        /// <summary>
+        /// Returns the current Runtime cache
+        /// </summary>
+        /// <remarks>
+        /// fixme - what is this? why not use RuntimeCache?
+        /// </remarks>
+        public IAppCache StaticCache { get; }
+
+        /// <summary>
+        /// Gets the runtime cache.
+        /// </summary>
+        /// <remarks>
+        /// <para>The runtime cache is the main application cache.</para>
+        /// </remarks>
+        public IAppPolicedCache RuntimeCache { get; }
```

Comment:
@zpqrtbnk  what is this name `IAppPolicedCache` ? what does this mean?

Reply:
we had `ICacheProvider` which becomes `IAppCache`, ie a cache for the application. Then we had `IRuntimeCacheProvider` but really, all it does is add support for Get/Insert with cache policies, to `ICacheProvide` - so have renamed `IAppPolicyCache` ie a cache for the application, that supports policies.

Trying to name the interface after what it does.